### PR TITLE
fix #2839

### LINKED ISSUE #2839: allow to search by sample_name or ebi sample accessions
This came up as part of creating a specific meta-analysis where the user only wanted to select a given set of samples from multiple studies, preparations and artifacts. The plan is to allow to search by _only_ those fields and display the results using the regular study listing page, which includes a + sign to display available studies.

@ElDeveloper mentioned that in the current implementation we add a full artifact vs. a given sample; thus we will need to implement this part too.

cc: @ElDeveloper and @wasade

### PR DESCRIPTION
**Pull request by antgonza**

### CODE REVIEW

**Review by wasade — APPROVED** [review]
A possible non blocking change for a little further code reduction, and a question about the `accessions` dict

File: qiita_db/metadata_template/sample_template.py [hunk 1]
```diff
@@ -332,3 +332,20 @@ def biosample_accessions(self, value):
...
+
+        Parameters
+        ----------
+        add_ebi_accessions : bool, optional
+            If this should add the ebi accessions
+        """
+        df = self._common_to_dataframe_steps()
+
+        if add_ebi_accessions:
+            accessions = self.ebi_sample_accessions
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_db/metadata_template/sample_template.py:347]: yes, AFAIK all samples in the prep/sample are represented.

File: qiita_db/metadata_template/sample_template.py [hunk 2]
```diff
@@ -332,3 +332,20 @@ def biosample_accessions(self, value):
...
+
+    def to_dataframe(self, add_ebi_accessions=False):
+        """Returns the metadata template as a dataframe
+
+        Parameters
+        ----------
+        add_ebi_accessions : bool, optional
+            If this should add the ebi accessions
+        """
+        df = self._common_to_dataframe_steps()
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_db/metadata_template/sample_template.py:344]: not strong feeling about leaving as is or changing ...

**Review by antgonza — COMMENTED** [review]
thank you @wasade

File: qiita_db/metadata_template/sample_template.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (wasade) [reply qiita_db/metadata_template/sample_template.py:344]: cool cool, then :shipit: i think :)

**Pull request merged**